server_test.cpp

```diff
@@ -0,0 +1,29 @@
+#include "gtest/gtest.h"
+#include "server.hpp"
+
+class ServerTest : public ::testing::Test {
+protected:
+	bool checkAccept(boost::system::error_code ec) {
+		http::server::server server_("127.0.0.1", "8080");
+		return server_.checkServer(ec);
+	}
+
+};
+
+TEST_F(ServerTest, ServerStatus) {
+	boost::system::error_code ec = boost::system::errc::make_error_code(boost::system::errc::success);
+    EXPECT_TRUE(checkAccept(ec));
+}
+
+TEST(ConnectionTest, ConnectionStatus) {
```

Comment:
Change this test. The connection_aborted error should be tested only when it is caught by our program. Change the parameters of server to throw a connection_aborted error. Update our server constructor so it throws errors.